### ISSUE TITLE
Braintree client returns a torn down instance from the cache

### ISSUE THREAD

**Issue by adorum** [braintree/braintree-web#372]
### General information

* SDK version: 3.34.0
* Environment: Sandbox
* Browser and OS: Chrome 66.0.3359.181 on Linux Mint 18.2

Braintree.client.create function returns an instance from cache after that instance was torn down.

Example Code:
```
class MyComponent extends React.Component {
  state = {
    braintreeClient: null,
  }

  componentDidMount() {
    braintree.client.create({
      authorization: "exampleToken",
    }).then(clientInstance => {
      this.setState({ braintreeClient: clientInstance });
    });
  }

  componentWillUnmount() {
    if (this.state.braintreeClient) {
      this.state.braintreeClient.teardown();
    }
  }

  render() {
  }
}
```
First time when to component is instantiated, a braintree client is instantiaded as well and saved to cache inside the library. When the react component componentWillUnmount hook is fired, I tern down the braintree client instance. Problem is that when I second time instantiate the react component, the braintree.client.create returns an instance from the cache - but unfortunatelly that instance was already torn down.

It throws an exception:
```
BraintreeError {
   name: "BraintreeError", 
   code: "METHOD_CALLED_AFTER_TEARDOWN", 
   message: "getVersion cannot be called after teardown.", type: "MERCHANT", details: undefined
}
```

I would assume that the teardown function deletes the client instance from the cache.

**Comment by Epreuve** [comment]
Hey @adorum thanks for bringing this to our attention.  I believe I know where the issue is that's causing this, and we're going to take a look at getting this fixed. 

I'll keep this issue open and post back once we're able to get a fix pushed for it.

**Comment by Epreuve** [comment]
An update on this, our next release should resolve this issue.  I'll comment here when its released.

**Comment by crookedneighbor** [comment]
This is fixed in 3.34.1. Thanks for your patience!
